opal/mca/accelerator/cuda/accelerator_cuda.c

```diff
@@ -109,8 +109,8 @@ static int accelerator_cuda_check_addr(const void *addr, int *dev_id, uint64_t *
         *flags |= MCA_ACCELERATOR_FLAGS_UNIFIED_MEMORY;
     }
     if (CUDA_SUCCESS != result) {
-        /* Cannot identify, return an error */
-        return -1;
+        /* This can happen if cuda is not initialized. Assume it is a host buffer. */
```

Comment:
It would be nice to distinguish between an actual failure from the call and "CUDA is not initialized" - is there a "Is CUDA initialized?" query in the CUDA API?

Reply:
I don't see a specific query, but we can test if the call returns CUDA_ERROR_NOT_INITIALIZED vs another error message. (based on https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__INITIALIZE.html)

Reply:
Sounds perfect :-)

Reply:
Done